serde_codegen/src/ser.rs

```diff
@@ -57,12 +57,43 @@ pub fn expand_derive_serialize(
     );
 
     let where_clause = &impl_generics.where_clause;
+    /*
```

Comment:
It wasn't intended for immediate merge. I left those two commented blocks as examples of how I attempted to add warning suppression attributes, in hopes someone would give good advice how to do this properly.